src/main/java/jenkins/scm/api/SCMSource.java

```diff
@@ -261,6 +262,52 @@ protected SCMRevision retrieve(@NonNull SCMHead head, @NonNull TaskListener list
     }
 
     /**
+     * Looks up a specific revision based on some SCM-specific set of permissible syntaxes.
+     * Delegates to {@link #retrieve(String, TaskListener)}.
+     * @param revision might be a branch name, a tag name, a cryptographic hash, a revision number, etc.
+     * @param listener the task listener (optional)
+     * @return a valid revision object corresponding to the argument, with a usable corresponding head, or null if malformed or not found
+     * @throws IOException if an error occurs while performing the operation.
+     * @throws InterruptedException if any thread has interrupted the current thread.
+     * @since FIXME
+     */
+    @CheckForNull
+    public final SCMRevision fetch(@NonNull String revision, @CheckForNull TaskListener listener)
+            throws IOException, InterruptedException {
+        return retrieve(revision, defaultListener(listener));
+    }
+
+    /**
+     * Looks up a specific revision based on some SCM-specific set of permissible syntaxes.
+     * The default implementation uses {@link #retrieve(SCMHeadObserver, TaskListener)}
+     * and looks for {@link SCMHead#getName} matching the argument (so typically only supporting branch names).
+     * @param revision might be a branch name, a tag name, a cryptographic hash, a revision number, etc.
+     * @param listener the task listener
+     * @return a valid revision object corresponding to the argument, with a usable corresponding head, or null if malformed or not found
+     * @throws IOException if an error occurs while performing the operation.
+     * @throws InterruptedException if any thread has interrupted the current thread.
+     * @since FIXME
+     */
+    @CheckForNull
+    protected SCMRevision retrieve(@NonNull final String revision, @NonNull TaskListener listener)
+            throws IOException, InterruptedException {
```

Comment:
Per above: option to throw a custom exception if revision format is invalid, perhaps?  Otherwise we just get a null.  

This has much greater benefits for custom implementations (GitSCM, I'm looking at you!) even if it is not as helpful for the default impl.